docs/architecture/0e-plans.md

```diff
@@ -0,0 +1,543 @@
+| `entropy.h` | `mbedtls_entropy_` | Private | [Internal eventually](#headers-that-will-become-internal-eventually) |
+| `error_common.h` | `mbedtls_*err*` | Private | [Internal eventually](#headers-that-will-become-internal-eventually) |
+| `gcm.h` | `mbedtls_gcm_` | Expose | [context types](#headers-with-context-types) |
+| `hkdf.h` | `mbedtls_hkdf_` | Delete | https://github.com/Mbed-TLS/mbedtls/issues/9150 |
+| `hmac_drbg.h` | `mbedtls_hmac_drbg_` | Private | [can be made fully private](#headers-that-can-be-made-fully-private) with a little work |
+| `lms.h` | `mbedtls_lms_` | Public | [no PSA equivalent](#cryptographic-mechanisms-with-no-psa-equivalent) |
+| `md.h` | `mbedtls_md_` | Expose | [context types](#headers-with-context-types), but likely [Public hash-only `md.h`](#public-hash-only-mdh) |
+| `md5.h` | `mbedtls_md5_` | Expose | [context types](#headers-with-context-types) |
+| `memory_buffer_alloc.h` | `mbedtls_memory_buffer_alloc_` | Public | [Platform headers](#platform-headers) |
+| `nist_kw.h` | `mbedtls_nist_kw_` | Public | [no PSA equivalent](#cryptographic-mechanisms-with-no-psa-equivalent) |
+| `oid.h` | `mbedtls_oid_` | Private | [OID interface](#oid-interface) |
+| `pem.h` | `mbedtls_pem_` | TBD | [Base64 and PEM](#base64-and-pem) |
+| `pk.h` | `mbedtls_pk_` | Public | [cryptography-adjacent](#cryptography-adjacent-headers) |
+| `pkcs12.h` | `mbedtls_pkcs12_` | Private | [can be made fully private](#headers-that-can-be-made-fully-private) |
+| `pkcs5.h` | `mbedtls_pkcs5_` | Private | [can be made fully private](#headers-that-can-be-made-fully-private) |
+| `platform.h` | `mbedtls_platform_` | Public | [Platform headers](#platform-headers) |
+| `platform_time.h` | `mbedtls_*time*` | Public | [Platform headers](#platform-headers) |
+| `platform_util.h` | `mbedtls_platform_` | Public | [Platform headers](#platform-headers) |
+| `poly1305.h` | `mbedtls_poly1305_` | Expose | [context types](#headers-with-context-types) |
+| `private_access.h` | N/A | Exposed | [Only for exposed macros ](#headers-that-remain-public-for-exposed-macros) |
+| `psa_util.h` | N/A | Public | [remains public](#headers-that-remain-public) but see [Private types in `psa_util.h`](#private-types-in-psa_utilh) |
+| `ripemd160.h` | `mbedtls_ripemd160_` | Expose | [context types](#headers-with-context-types) |
+| `rsa.h` | `mbedtls_rsa_` | Private | [can be made fully private](#headers-that-can-be-made-fully-private) with a little work (TODO) |
+| `sha1.h` | `mbedtls_sha1_` | Expose | [context types](#headers-with-context-types) |
+| `sha256.h` | `mbedtls_sha256_` | Expose | [context types](#headers-with-context-types) |
+| `sha3.h` | `mbedtls_sha3_` | Expose | [context types](#headers-with-context-types) |
+| `sha512.h` | `mbedtls_sha512_` | Expose | [context types](#headers-with-context-types) |
+| `threading.h` | `mbedtls_threading_` | Public | [Platform headers](#platform-headers) |
```

Comment:
This list is missing `timing.h` I believe

Reply:
Is it, though? I'm not seeing any `timing.h` in `drivers/builtin/include/mbedtls/` in today's development, and apparently it was already not there in e35785c3c15924d1a7024a4b8695ce32ed186bf9 (when the repo was created).

Reply:
It was briefly in crypto, but already moved back to mbedtls before the repo split.